src/libraries/Microsoft.Extensions.DependencyModel/src/EnvironmentWrapper.cs

```diff
@@ -14,6 +16,11 @@ internal sealed class EnvironmentWrapper : IEnvironment
 
         public object? GetAppContextData(string name) => AppDomain.CurrentDomain.GetData(name);
 
-        public bool IsWindows() => RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        public bool IsWindows() =>
+#if NETFRAMEWORK
+            Environment.OSVersion.Platform == PlatformID.Win32NT;
```

Comment:
The .NET Framework assembly might be used by folks targeting mono or unity which may be non-windows.

That said in this particular case the library is later probing environment variables and will fall-back if they don't exist so it might be OK to just remove this check entirely it if we don't expect folks to define those variables on non-windows; https://source.dot.net/#Microsoft.Extensions.DependencyModel/IEnvironment.cs,30a616f27e3ca3a6,references

Reply:
Also, given that this library is for reading the deps file, it might be very unlikely that folks are using it on mono.

Reply:
While that's unlikely, it's still plausible, especially when someone uses legacy mono. I want to avoid a potential breaking change for existing consumers and reading from Environment.OSVersion is simple enough.